### ISSUE TITLE
possible to change or add arguments for kubelet?

### ISSUE THREAD

**Issue by ltjsjyyy** [rancher/rancher#14041]
possible to add additional startup arguments for kubelet? such as "--provider-id=XXXXX" and "--hostname-override=XXXXX"?

**Comment by LinAnt** [comment]
This can be done through the RKE config. It is currently only possible through the rancher API. You can even do it through the API web UI if you specify that gke, aws and azure configs are null.
Here is the config we are currently using for istio 0.8.0 support:

```{
  "addonJobTimeout": 30,
  "authentication": { "strategy": "x509", "type": "/v3/schemas/authnConfig" },
  "authorization": { "type": "/v3/schemas/authzConfig" },
  "bastionHost": { "sshAgentAuth": false, "type": "/v3/schemas/bastionHost" },
  "cloudProvider": { "type": "/v3/schemas/cloudProvider" },
  "ignoreDockerVersion": true,
  "ingress": { "provider": "nginx", "type": "/v3/schemas/ingressConfig" },
  "kubernetesVersion": "v1.10.1-rancher2-1",
  "network": {
    "flannelNetworkProvider": {
      "iface": "",
      "type": "/v3/schemas/flannelNetworkProvider"
    },
    "plugin": "flannel",
    "type": "/v3/schemas/networkConfig"
  },
  "services": {
    "etcd": {
      "extraArgs": { "election-timeout": "5000", "heartbeat-interval": "500" },
      "snapshot": false,
      "type": "/v3/schemas/etcdService"
    },
    "kubeApi": {
      "podSecurityPolicy": false,
      "extraArgs": {
        "admission-control":
            "ServiceAccount,NamespaceLifecycle,LimitRanger,PersistentVolumeLabel,DefaultStorageClass,ResourceQuota,DefaultTolerationSeconds,MutatingAdmissionWebhook,ValidatingAdmissionWebhook",
        "runtime-config": "admissionregistration.k8s.io/v1alpha1"
      },
      "type": "/v3/schemas/kubeAPIService"
    },
    "kubeController": { 
        "type": "/v3/schemas/kubeControllerService",
        "extraArgs": {
        "cluster-signing-cert-file": "/etc/kubernetes/ssl/kube-ca.pem",
        "cluster-signing-key-file": "/etc/kubernetes/ssl/kube-ca-key.pem"
        }
    },
    "kubelet": { "failSwapOn": false, "type": "/v3/schemas/kubeletService" },
    "kubeproxy": { "type": "/v3/schemas/kubeproxyService" },
    "scheduler": { "type": "/v3/schemas/schedulerService" },
    "type": "/v3/schemas/rkeConfigServices"
  },
  "sshAgentAuth": false,
  "type": "/v3/schemas/rancherKubernetesEngineConfig"
}
```

**Comment by superseb** [comment]
This is covered in https://github.com/rancher/rancher/issues/13076.

See above comment, https://gist.github.com/superseb/0c5e4015985f470853f0a74427c306c7, or https://github.com/JasonvanBrackel/metrics-server-on-rancher-2.0.2 for more examples how to do it until a version is released with #13076

**Comment by ltjsjyyy** [comment]
@superseb 
thanks for your reply, it works for a static value of extra arguments, but how can i set a expression for the hosts? in my example, provider-id could be a hostname of the host.
